### ISSUE TITLE
Build error with setuptools 0.68.0

### ISSUE THREAD

**Issue by phillipuniverse** [yaml/pyyaml#723]
I think this might be caused from an inadvertent upgrade in my system of [virtualenv from 20.23.1 to 20.24.0](https://virtualenv.pypa.io/en/latest/changelog.html#v20-24-0-2023-07-14), which included an upgrade of setuptools from [67.8 to 68](https://setuptools.pypa.io/en/latest/history.html#v68-0-0).

This was with Alpine (and Python 3.11.4) but the problem seems to be setuptools 68:

```
  ChefBuildError

[2023-07-17T15:25:18.341Z]   Backend subprocess exited when trying to invoke get_requires_for_build_wheel
  
  running egg_info
  writing lib/PyYAML.egg-info/PKG-INFO
  writing dependency_links to lib/PyYAML.egg-info/dependency_links.txt
  writing top-level names to lib/PyYAML.egg-info/top_level.txt
  Traceback (most recent call last):
    File "/usr/local/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
      main()
    File "/usr/local/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 335, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/usr/local/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
      return hook(config_settings)
             ^^^^^^^^^^^^^^^^^^^^^
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=['wheel'])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
      self.run_setup()
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/build_meta.py", line 338, in run_setup
      exec(code, locals())
    File "<string>", line 288, in <module>
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/__init__.py", line 107, in setup
      return distutils.core.setup(**attrs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
      return run_commands(dist)
             ^^^^^^^^^^^^^^^^^^
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
      dist.run_commands()
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
      self.run_command(cmd)
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/dist.py", line 1234, in run_command
      super().run_command(command)
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
      cmd_obj.run()
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 314, in run
      self.find_sources()
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 322, in find_sources
      mm.run()
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 551, in run
      self.add_defaults()
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 589, in add_defaults
      sdist.add_defaults(self)
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/command/sdist.py", line 104, in add_defaults
      super().add_defaults()
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/_distutils/command/sdist.py", line 251, in add_defaults
      self._add_defaults_ext()
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/_distutils/command/sdist.py", line 336, in _add_defaults_ext
      self.filelist.extend(build_ext.get_source_files())
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "<string>", line 204, in get_source_files
    File "/tmp/tmp0l7lnl_n/.venv/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 107, in __getattr__
      raise AttributeError(attr)
  AttributeError: cython_sources
  

  at /usr/local/lib/python3.11/site-packages/poetry/installation/chef.py:147 in _prepare
      143│ 
      144│                 error = ChefBuildError("\n\n".join(message_parts))
      145│ 
      146│             if error is not None:
    → 147│                 raise error from None
      148│ 
      149│             return path
      150│ 
      151│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:

Note: This error originates from the build backend, and is likely not a problem with poetry but with pyyaml (6.0) not supporting PEP 517 builds. You can verify this by running 'pip wheel --use-pep517 "pyyaml (==6.0)"'.

```

**Comment by phillipuniverse** [comment]
Looks like the error happens at https://github.com/yaml/pyyaml/blob/957ae4d495cf8fcb5475c6c2f1bce801096b68a5/setup.py#L203-L206

**Comment by mcdonnnj** [comment]
https://github.com/cython/cython/issues/4568

**Comment by phillipuniverse** [comment]
@mcdonnnj so since this is a CPython issue, does that mean won't be fixed til 3.11.4?

**Comment by mcdonnnj** [comment]
@phillipuniverse Cython != CPython

**Comment by wesleykendall** [comment]
@phillipuniverse I also get this error, but as a result of using `pip-tools`. It happens for both versions of setuptools you mentioned.

Are you also using pip-tools when getting this error?

**Comment by mcdonnnj** [comment]
This is pretty much a duplicate of https://github.com/yaml/pyyaml/issues/601 with https://github.com/yaml/pyyaml/pull/702 intending to stopgap.

**Comment by Kaelten** [comment]
I'm just using poetry and hitting this error along with a bunch of other's.  #601 is getting a lot of traffic related to this too.  

Does anyone know of a workaround that'd work with poetry?

**Comment by dennistruemper** [comment]
I was trying to install awscli in version 1.22.11 and it needs PyYAML too, the version 5.3.1 was working fine, so maybe this is a help for someone with a similar problem

**Comment by wesleykendall** [comment]
@dennistruemper thanks, this is what ultimately fixed it for me. Seems like pyyaml==5.4.* causes this. Hopefully awscli will support 6.* soon

**Comment by lordoffreaks** [comment]
> I was trying to install awscli in version 1.22.11 and it needs PyYAML too, the version 5.3.1 was working fine, so maybe this is a help for someone with a similar problem

I can confirm using 5.3.1 works for me

**Comment by AlexDld** [comment]
yes I think using 5.3.1 is the only way

**Comment by phillipuniverse** [comment]
> This is pretty much a duplicate of https://github.com/yaml/pyyaml/issues/601 with https://github.com/yaml/pyyaml/pull/702 intending to stopgap.

@mcdonnnj agree, I'll close this one to consolidate conversation.

**Comment by tommyjcarpenter** [comment]
the title of this issue is wrong, its not 0.68.0

**Comment by trend-mars** [comment]
@tommyjcarpenter, I think this problem occurs in other setuptools versions as well. I got this error from updating setuptools from 59.6.0 to 67.8.0.

But after I fixed the PyYAML version at 5.3.1. Problem solved.

![Screenshot 2023-07-19 at 9 36 56 AM](https://github.com/yaml/pyyaml/assets/138850294/37895a4a-fe7d-49a0-96e0-898b9c9f7bab)
![Screenshot 2023-07-19 at 9 37 10 AM](https://github.com/yaml/pyyaml/assets/138850294/8b6a20ae-9676-4464-a67f-ca046b959af5)

**Comment by Pudding124** [comment]
@tommyjcarpenter, I think this problem occurs in other setuptools versions as well. I got this error from updating setuptools from 59.6.0 to 67.8.0.

But after I fixed the PyYAML version at 5.3.1. Problem solved.
![Screenshot 2023-07-19 at 9 36 56 AM](https://github.com/yaml/pyyaml/assets/19733850/2c0cadb4-ab26-4a1c-83dd-d5b308357a58)
![Screenshot 2023-07-19 at 9 37 10 AM](https://github.com/yaml/pyyaml/assets/19733850/8d59c19c-7d45-467f-ad53-33be9415a9e5)

**Comment by Mishap111** [comment]
Christopher Mestas
